packages/editor/src/utils/get-template-info.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * WordPress dependencies
+ */
+import { layout } from '@wordpress/icons';
+/**
+ * Internal dependencies
+ */
+import { getTemplatePartIcon } from './get-template-part-icon';
+const EMPTY_OBJECT = {};
+
+/**
+ * Helper function to retrieve the corresponding template info for a given template.
+ * @param {Object} params
+ * @param {Array}  params.templateTypes
+ * @param {Array}  params.templateAreas
+ * @param {Object} params.template
+ */
+export const getTemplateInfo = ( params ) => {
```

Comment:
Ultimately, this looks like something that belongs in "core-data" but ok to keep I guess here for now.

Reply:
Also, I'm not really sure if a big utility like that is better than smaller ones `getTemplateTitle` `getTemplateIcon` and `getTemplateDescription`. That said, feels like something to be explored separately.